crates/pathfinder/src/storage/merkle_tree.rs

```diff
@@ -181,20 +279,26 @@ impl<'a> RcNodeStorage<'a> {
     ///
     /// None of the [RcNodeStorage] functions rollback on failure. This means that if any error
     /// is encountered, the transaction should be rolled back to prevent database corruption.
-    pub fn open(table: String, transaction: &'a Transaction<'_>) -> anyhow::Result<Self> {
-        transaction.execute(
-            &format!(
-                r"CREATE TABLE IF NOT EXISTS {}(
-                    hash        BLOB PRIMARY KEY,
-                    data        BLOB,
-                    ref_count   INTEGER
-                )",
-                &table
-            ),
-            [],
-        )?;
+    pub fn open(table: &str, transaction: &'tx Transaction<'tx>) -> anyhow::Result<Self> {
+        let queries = if table == "tree_global" {
+            let q = GLOBAL.borrow();
+            // this assertion exists to prove that the reborrowing works.
+            debug_assert!(matches!(q.create, Cow::Borrowed(_)));
+            q
+        } else if table == "tree_contracts" {
+            CONTRACTS.borrow()
+        } else {
+            Queries::format(table)
+        };
```

Comment:
Could a `match table.as_str()` be easier to understand maybe?

Reply:
Table is already a str, unsure how it could be easier to undestand but can change.